src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -812,7 +815,12 @@ def sql_query(self) -> str:
     def schema_query(self) -> str:
         """The first operand decide the column attributes of a query with set operations.
         Refer to https://docs.snowflake.com/en/sql-reference/operators-query.html#general-usage-notes"""
-        return self.set_operands[0].selectable.schema_query
+        attributes = self.set_operands[0].selectable.snowflake_plan.attributes
+        sql = f"({schema_value_statement(attributes)})"
+        for i in range(1, len(self.set_operands)):
+            attributes = self.set_operands[i].selectable.snowflake_plan.attributes
+            sql = f"{sql}{self.set_operands[i].operator}({schema_value_statement(attributes)})"
```

Comment:
I'm just curious: what is the operator of the first operand? Is it None?

Reply:
It is `None` indeed. If you expand a few lines above, the new `schema_query` generation follows the same pattern as `sql_query` generation.